src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -144,52 +210,82 @@ sealed trait LazyModuleImpLike extends RawModule
     val allDangles = nodeDangles ++ childDangles
+    /** For [[allDangles]] which originate from the same [[Dangle.source]], group them together into [[pairing]]. */
     val pairing = SortedMap(allDangles.groupBy(_.source).toSeq:_*)
+    /** For each set of [[Dangle]]s where there are sets of 2 with the same source and different flipped,
+      * ensure that they can be connected in a source/sink pair, and then make the connection.
+      * For these sets, mark them as [[done]]. */
     val done = Set() ++ pairing.values.filter(_.size == 2).map { case Seq(a, b) =>
       require (a.flipped != b.flipped)
+      /** @todo <> in chisel3 makes directionless connection. */
       if (a.flipped) { a.data <> b.data } else { b.data <> a.data }
       a.source
     }
+    /** Find all [[Dangle]]s which are still not connected. These will end up as [[AutoBundle]] [[IO]] ports on the module.*/
     val forward = allDangles.filter(d => !done(d.source))
+    /** Generate [[AutoBundle]] io from [[forward]]. */
     val auto = IO(new AutoBundle(forward.map { d => (d.name, d.data, d.flipped) }:_*))
+    /** Pass the [[Dangle]]s which remained and were used to generate the [[AutoBundle]] I/O ports up to the [[parent]] [[LazyModule]]*/
     val dangles = (forward zip auto.elements) map { case (d, (_, io)) =>
       if (d.flipped) { d.data <> io } else { io <> d.data }
       d.copy(data = io, name = wrapper.suggestedName + "_" + d.name)
     }
+    /** Push all [[LazyModule.inModuleBody]] to [[chisel3.internal.Builder]]. */
     wrapper.inModuleBody.reverse.foreach { _() }
+    /** Return [[IO]] and [[Dangle]] of this [[LazyModuleImp]]. */
     (auto, dangles)
   }
 
+  /** Ask each [[BaseNode]] in [[wrapper.nodes]] to call [[BaseNode.finishInstantiate]]
+    * Note: There are 2 different `finishInstantiate` methods:
+    *   [[LazyModuleImp.finishInstantiate]] and [[BaseNode.finishInstantiate]],
+    *   the former is a wrapper to the latter.
+    * */
   protected[diplomacy] def finishInstantiate() {
     wrapper.nodes.reverse.foreach { _.finishInstantiate() }
   }
 }
 
 class LazyModuleImp(val wrapper: LazyModule) extends MultiIOModule with LazyModuleImpLike {
+  /** Instantiate hardware of this `Module`. */
   val (auto, dangles) = instantiate()
 }
 
 class LazyRawModuleImp(val wrapper: LazyModule) extends RawModule with LazyModuleImpLike {
-  // These wires are the default clock+reset for all LazyModule children
-  // It is recommended to drive these even if you manually shove most of your children
-  // Otherwise, anonymous children (Monitors for example) will not be clocked
+  /** These wires are the default clock+reset for all LazyModule children.
+    * It is recommended to drive these even if you manually drive the [[clock]] and [[reset]] of all of the [[LazyRawModuleImp]] children.
+    * Otherwise, anonymous children ([[Monitor]]s for example) will not have their [[clock]] and/or [[reset]] driven properly.
+    * */
```

Comment:
I'm not suure how scaladoc will work, but wouldn't we want an explicit scaladoc comment for each of `childClock` and `childReset`?